src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+private:
+    QString m_keyName;
```

Comment:
~~`SettingStorage::loadValue(const char*,...)` and `SettingStorage::storeValue(const char*, ...)` would be a valuable addition.~~ No, with `QVariantMap` inside this just does not make any sense.